src/main/java/org/generationcp/ibpworkbench/security/WorkbenchAuthenticationSuccessHandler.java

```diff
@@ -57,6 +58,16 @@ public void onAuthenticationSuccess(final HttpServletRequest request, final Http
 		// The projectId and token will be populated later when a program is opened/loaded.
 		ContextUtil.setContextInfo(request, user.getUserid(), null, null, shouldShowReleaseNote);
 
+		UserInfo userInfo = this.userService.getUserInfo(user.getUserid());
```

Comment:
@nahuel-soldevilla we should move the logic of handling the hibernate entity to middleware
UserService is exposing hibernate entities in most of the cases. I think will it be better to create a new function in UserService that receives the userId and decides wheter to insert or update userInfo